utils/src/util.c

```diff
@@ -0,0 +1,63 @@
+#define _GNU_SOURCE
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <string.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <wordexp.h>
+
+#include "util.h"
+
+static int open_path(char *path, int flags)
+{
+	int ret;
+
+	ret = open(path, flags);
+	if (ret < 0) {
+		ret = -errno;
+		fprintf(stderr, "failed to open '%s': %s (%d)\n",
+			path, strerror(errno), errno);
+		return ret;
+	}
+
+	return ret;
+}
+
+/*
+ * 1. if path option given, use that
+ * 2. if env var, use that
+ * 3. if cwd is in a scoutfs fs, use that
+ * 4. else error
+ */
+int get_path(char *path, int flags)
+{
+	char *env_path;
+	char *cur_dir_path;
+	int ret;
+
+	if (path) {
+		wordexp_t exp_result;
+		wordexp(path, &exp_result, 0);
+		return open_path(exp_result.we_wordv[0], flags);
+	}
+
+	env_path = getenv("SCOUTFS_PATH");
+	if (env_path)
+		return open_path(path, flags);
+
+	cur_dir_path = get_current_dir_name();
+	if (!cur_dir_path) {
+		ret = -errno;
+		return ret;
+	}
+
+	ret = open_path(cur_dir_path, flags);
+	free(cur_dir_path);
+
+	// TODO: check this is within a scoutfs mount?
```

Comment:
We already can use CWD, it'll just cause an error if the CWD isn't within a ScoutFS mount. The only benefit of this would be we could print a more specific diagnostic message: "current working directory is not within a ScoutFS mount" rather than "inappropriate ioctl for device (25)". I'm going to leave this as a TODO.

Reply:
Sounds good.